tutorials/modules/ROOT/pages/create-materialized-view-from-kafka.adoc

```diff
@@ -0,0 +1,349 @@
++
+```sql
+INSERT INTO companies VALUES
+(1, 'SORG', 'Example Startup Organization', 100000),
+(2, 'EORG', 'Example Enterprise Organization', 5000000);
+```
+
+. Merge results from the `companies` map and `trades` topic so you can see the company name that's associated with each ticker.
++
+```sql
+SELECT trades.ticker, companies.company, trades.amount
+FROM trades
+JOIN companies
+ON companies.ticker = trades.ticker;
+```
++
+You should see that Hazelcast is executing the streaming query.
++
+```
++------------+-------------------------------+--------------+
+|ticker      |company                        |amount        |
++------------+-----------+-------------------+--------------|
+|SORG        |Example Startup Organization   |10            |
+|EORG        |Example Enterprise Organization|20            |
+```
+
+. Click *Stop Query*.
+
+
+== Step 4. Create a Materialized View
+
+You can set up an automated job to continuously run the streaming query and cache the results in a Hazelcast map.
+
+. Open the SQL browser in Management Center.
+
+. Create a mapping to a new map in which to cache your streaming query results. This is your materialized view.
++
+```sql
+CREATE MAPPING trade_map (
+__key BIGINT,
+ticker VARCHAR,
+company VARCHAR,
+amount BIGINT)
+TYPE IMap
+OPTIONS (
+'keyFormat'='bigint',
+'valueFormat'='json-flat');
+```
+
+. Submit a job to your cluster that will monitor your `trade` topic for changes and store them in a map. The processing guarantee tells Hazelcast to save the current offsets so that the cluster can resume the job even if the cluster restarts.
++
+```sql
+CREATE JOB ingest_trades
+OPTIONS (
+  'processingGuarantee' = 'exactlyOnce',
+) AS
+SINK INTO trade_map
+SELECT trades.id, trades.ticker, companies.company, trades.amount
+FROM trades
+JOIN companies
+ON companies.ticker = trades.ticker;
+```
++
+A job will run indefinitely until it is explicitly canceled or the cluster is shut down. Even if you exit the command prompt, the job will continue running on the cluster.
+
+. List your job to make sure that it was successfully submitted.
++
+```sql
+SHOW JOBS;
+```
++
+You should see a job called `ingest_trades`.
++
+```
++--------------------+
+|name                |
++--------------------+
+|ingest_trades       |
++--------------------+
+```
+
+. Query your materialized view to see that results have been added to it.
++
+```sql
+SELECT * FROM trade_map;
+```
++
+You should see that the query results are being stored in your map.
++
+```
++---------+---------+---------------------------------+------------+
+|       id|ticker   |   company                       |      amount|
++---------+---------+---------------------------------+------------+
+|        2|EORG     |Example Enterprise Organization  |          20|
+|        1|SORG     |Example Startup Organization     |          10|
++---------+---------+----------+----------------------+------------+
+```
++
+Leave this query running.
```

Comment:
Hm, this query is not running, it was executed and finished. Maybe we can remove this line?